ast/compile.go

```diff
@@ -1348,6 +1449,7 @@ type queryCompiler struct {
 	after                map[string][]QueryCompilerStageDefinition
 	unsafeBuiltins       map[string]struct{}
 	comprehensionIndices map[*Term]*ComprehensionIndex
+	schema               interface{}
```

Comment:
The `checkTypes` function on the query compiler should be extended to do the same thing as the Compiler.

Reply:
Done